### ISSUE TITLE
case/noexec.code probably should be an error

### ISSUE THREAD

**Issue by bondbenbond** [zqianem/corelang-test-files#3]
https://github.com/zqianem/corelang-test-files/commit/115efbfe301c0c5c692c8323f74c47a43abe5479

```
program
  int x;
begin
  x:=1;
  case x of
    0:y
    else 4
  end;
  output x;
end
4
```

I understand the logic of it not throwing an error for 'y', but seems like it would still be an undeclared variable that would be caught during parsing. Thoughts?

https://piazza.com/class/jza8gjfouuh5vt?cid=110

**Error handling of undeclared variables**
Are we allowed to check to check the validity of variables during the executor rather than parser? By validity, I'm referring to ensuring all variables used in statements were declared beforehand. 

> These checks should be performed during parsing, or immediately after parsing.

**Comment by zqianem** [comment]
Thanks for opening this issue! Sorry for this late-as-to-this-being-potentially-irrelevant response. I do think this is an interesting issue to discuss, however.

Based purely on the linked discussion, you are right in that this would be an error that could and should be checked for during parsing, or immediately afterwards.

Personally, I'm a bit biased in seeing it that way just because of the way I implemented my interpreter, in which the parser's only job was to generate a parse tree based on the context-free grammar, and the executor's only job was to attempt to traverse the parse tree to actually run the program. I didn't include the sort of static checks that would catch this sort of thing before execution in attempt to mimic interpreted languages like Python and Ruby, in which this kind of semantic error would only interrupt execution if the section of code containing it was actually executed, even if it was something obvious like `x = 1/0`.

A tangent: compared to other interpreted languages, CORE is bit strange because it has the C-like concept of variables having to be explicitly declared before they can be used, which I believe the only benefit of which is to indicate to the compiler how much memory set aside at certain point, like for a stack frame or something. Obviously CORE doesn't have a compiler (unless that's the next project?), so the idea of having to "pre-declare" variables at all seems somewhat ill-fitting to the language, especially since the data structure used to represent the symbol table in the interpreter is likely dynamic in nature anyhow and would not benefit from the knowledge that declaring variables beforehand provides.

So in short, you're right in that this is probably not valid CORE code, but I'm going to be stubborn and say it should run anyway* because I'm too lazy to change things now past the deadline :stuck_out_tongue:.

To actually address the issue, we could add some sort of toggle to the shell script to filter out these test cases from the main ones, or add some note in the documentation about this. Feel free to open a PR for this purpose. On the other hand, just keeping this issue open will probably suffice, in the interest maximum laziness.


*There is some vague precedence for this in some other tests cases like `generic-fun/fib2000`, where `Integer.MAX_VALUE` is definitely exceeded.

**Comment by bondbenbond** [comment]
Great points. Thanks for the making this, it was super helpful! :)
